packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -49,41 +45,84 @@ function NavigationMenuItemEdit( {
 	isParentOfSelectedBlock,
 	setAttributes,
 	insertMenuItemBlock,
+	fetchSearchSuggestions,
 } ) {
+	const { label, opensInNewTab, title, url } = attributes;
+	const link = title ? { title, url } : null;
 	const [ isLinkOpen, setIsLinkOpen ] = useState( false );
-	const [ isEditingLink, setIsEditingLink ] = useState( false );
-	const [ urlInput, setUrlInput ] = useState( null );
+	const [ wasClosedByLinkControl, setWasClosedByLinkControl ] = useState( false );
+
+	/**
+	 * It's a kind of hack to handle closing the LinkControl popover
+	 * clicking on the ToolbarButton link.
+	 */
+	useEffect( () => {
+		if ( ! isSelected ) {
+			setIsLinkOpen( false );
+			setWasClosedByLinkControl( false );
+		}
+		return () => {
+			setIsLinkOpen( false );
```

Comment:
This function is called when the component is going to be unmounted. If the component is being unmounted I think setting the state is irrelevant and we may able to remove the return function.